src/system_contract.rs

```diff
@@ -25,10 +28,38 @@ pub enum SystemContract {
     /// * Transaction::keys[0] - source
     /// * Transaction::keys[1] - destination
     Move { tokens: i64 },
+    /// Load a contract
+    /// contract_id - id to associate this contract
+    /// nanme - file path of the contract to load
+    Load { contract_id: Pubkey, name: String },
+    /// Call a contract
+    /// contract_id - contract to call
+    /// data - contract specific data
+    Call { contract_id: Pubkey, data: Vec<u8> },
 }
 
 pub const SYSTEM_CONTRACT_ID: [u8; 32] = [0u8; 32];
 
+#[derive(Debug)]
+pub struct AccountInfo<'a> {
+    pub key: &'a Pubkey,
+    pub account: &'a mut Account,
```

Comment:
Hmm, how about dumping the references here?

Reply:
Not sure how to how since they are both borrowed by `SystemContract::process_transaction()`.  Are you suggesting a rework of `SystemContract`?